File: app/src/test/java/com/softdev/smartraysam/carfeeds/PresenterTest.java
package com.softdev.smartraysam.carfeeds;

import android.content.Context;
import android.util.Log;

import com.softdev.smartraysam.carfeeds.Core.GetDataContract;
import com.softdev.smartraysam.carfeeds.Core.Intractor;
import com.softdev.smartraysam.carfeeds.Core.Presenter;
import com.softdev.smartraysam.carfeeds.model.carModel;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

import java.util.List;

import static org.junit.Assert.*;

public class PresenterTest  {
    @Mock
    private GetDataContract.View mGetDataView;

    @Mock
    private GetDataContract.Interactor mInteractor;
    @Mock
    private  GetDataContract.onGetDataListener mListener;

    private GetDataContract.Presenter presenter;
    Context context;
    String url="https://s3-us-west-2.amazonaws.com";
    String msg;
    @Mock
    List<carModel> allData;
    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        mInteractor= new Intractor(mListener);
        presenter = new Presenter(mGetDataView);
    }
    @Test
    public void getDataFromURL( ) {
        mInteractor.initRetrofitCall(context,url);
        mGetDataView.onStart(msg);
    }
    @Test
    public void onSuccess() {
        mGetDataView.onGetDataSuccess(msg,allData);
        Log.v("onSuccess",msg);
    }

    @Test
    public void onFailure() {
        mGetDataView.onGetDataFailure(msg);
        Log.v("onFailure",msg);
    }
}

File: app/src/test/java/com/softdev/smartraysam/carfeeds/PresenterTest.kt
package com.softdev.smartraysam.carfeeds

import android.content.Context
import android.util.Log

import com.softdev.smartraysam.carfeeds.contract.GetDataContract
import com.softdev.smartraysam.carfeeds.interactor.Intractor
import com.softdev.smartraysam.carfeeds.Presenter.Presenter
import com.softdev.smartraysam.carfeeds.model.carModel

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class PresenterTest {
    @Mock
    private val mGetDataView: GetDataContract.View? = null

    @Mock
    private var mInteractor: GetDataContract.Interactor? = null
    @Mock
    private val mListener: GetDataContract.onGetDataListener? = null

    private var presenter: GetDataContract.Presenter? = null
    internal var context: Context? = null
    internal var url = "https://s3-us-west-2.amazonaws.com"
    internal var msg: String? = null
    @Mock
    internal var allData: List<carModel>? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        MockitoAnnotations.initMocks(this)
        mInteractor = Intractor(mListener!!)
        presenter = Presenter(mGetDataView!!)
    }

    @Test
    fun getDataFromURL() {
        msg?.let { nonNullMessage ->
            mInteractor!!.initRetrofitCall(context!!, url)
            mGetDataView!!.onStart(msg!!)
        }

    }

    @Test
    fun onSuccess() {
        msg?.let { nonNullMessage ->
            mGetDataView!!.onGetDataSuccess(msg!!, allData!!)
            Log.v("onSuccess", msg)
        }

    }

    @Test
    fun onFailure() {
        msg?.let { nonNullMessage ->
            mGetDataView!!.onGetDataFailure(msg!!)
            Log.v("onFailure", msg)
        }
    }
}